File: application/backend-credit-card-enrollment/backend-c#/CreditCardEnrollment/Domain/Product/Events/ProductDefined.cs
using CreditCardEnrollment.Common.Events;

namespace CreditCardEnrollment.Domain.Product.Events;

public class ProductDefined : CreationEvent
{
    public required string Name { get; init; }
    public required int InterestInBasisPoints { get; init; }
    public required int AnnualFeeInCents { get; init; }
    public required string PaymentCycle { get; init; }
    public required int CreditLimitInCents { get; init; }
    public required int MaxBalanceTransferAllowedInCents { get; init; }
    public required string Reward { get; init; }
    public required string CardBackgroundHex { get; init; }
}

File: application/backend-credit-card-enrollment/backend-typescript/src/creditCard/product/event/ProductDefined.ts
import { Event } from '../../../common/event/Event';

export class ProductDefined extends Event {
    constructor(
        eventId: string,
        aggregateId: string,
        aggregateVersion: number,
        correlationId: string,
        causationId: string,
        recordedOn: Date,
        public readonly name: string,
        public readonly interestInBasisPoints: number,
        public readonly annualFeeInCents: number,
        public readonly paymentCycle: string,
        public readonly creditLimitInCents: number,
        public readonly maxBalanceTransferAllowedInCents: number,
        public readonly reward: string,
        public readonly cardBackgroundHex: string
    ) {
        super(eventId, aggregateId, aggregateVersion, correlationId, causationId, recordedOn);
    }
}
